### PR TITLE
Split the BBE, send-and-receive-emails into three BBEs for clarity

### PR DESCRIPTION
**Pull request by Maninda**
Split the BBE, `send-and-receive-emails` into the following 3 BBEs.
1. send-email
2. receive-email-using-client
3. receive-email-using-listener

### CODE REVIEW

File: examples/receive-email-using-client/receive_email_using_client.bal [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    // error is returned.
+    email:Message? emailResponse = check popClient->receiveEmailMessage();
+
+    if (emailResponse is email:Message) {
+        io:println("POP client received an email.");
+        io:println("Email Subject: ", emailResponse.subject);
+        io:println("Email Body: ", emailResponse.body);
+    // When no emails are available in the server, `()` is returned.
+    } else {
+
```
Comment (lafernando) [inline examples/receive-email-using-client/receive_email_using_client.bal:22]: Remove extra line.

File: examples/receive-email-using-client/receive_email_using_client.bal [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    // error is returned.
+    emailResponse = check imapClient->receiveEmailMessage();
+
+    if (emailResponse is email:Message) {
+        io:println("IMAP client received an email.");
+        io:println("Email Subject: ", emailResponse.subject);
+        io:println("Email Body: ", emailResponse.body);
+    // When no emails are available in the server, `()` is returned.
+    } else {
+
```
Comment (lafernando) [inline examples/receive-email-using-client/receive_email_using_client.bal:43]: Remove extra line.

File: examples/receive-email-using-client/receive_email_using_client.description [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// The Email Connector is used to receive (with POP3 or IMAP4) emails
```
Comment (lafernando) [inline examples/receive-email-using-client/receive_email_using_client.description:1]: Shall we say "The email client" ..

File: examples/receive-email-using-listener/receive_email_using_listener.description [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// The Email Connector is used to receive (with POP3 or IMAP4) emails
```
Comment (lafernando) [inline examples/receive-email-using-listener/receive_email_using_listener.description:1]: Shall we say "The email listener" ..

**Review by lafernando — CHANGES_REQUESTED** [review]

File: examples/receive-email-using-client/receive_email_using_client.description [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+// using the SSL or STARTTLS protocols. This sample includes receiving
+// emails with default configurations over SSL using default
```
Comment (praneesha) [inline examples/receive-email-using-client/receive_email_using_client.description:3]:
```suggestion
// emails with default configurations over SSL using the default
```

File: examples/receive-email-using-client/receive_email_using_client.metatags [hunk 6]
```diff
@@ -0,0 +1,2 @@
+description: This is a BBE on receiving emails. It has client functionality for email receiving.
```
Comment (praneesha) [inline examples/receive-email-using-client/receive_email_using_client.metatags:1]:
```suggestion
description: This is a BBE on receiving emails. It has the client functionality for receiving emails.
```

File: examples/receive-email-using-listener/receive_email_using_listener.description [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+// using the SSL or STARTTLS protocols. This sample includes receiving
+// emails from a listener with default configurations over SSL using
+// default ports.<br/><br/>
```
Comment (praneesha) [inline examples/receive-email-using-listener/receive_email_using_listener.description:4]:
```suggestion
// the default ports.<br/><br/>
```

File: examples/receive-email-using-listener/receive_email_using_listener.metatags [hunk 8]
```diff
@@ -0,0 +1,2 @@
+description: This is a BBE on receiving emails. It has a listener based functionality for receiving emails.
```
Comment (praneesha) [inline examples/receive-email-using-listener/receive_email_using_listener.metatags:1]:
```suggestion
description: This BBE is about receiving emails. It has a listener-based functionality for receiving emails.
```

File: examples/send-email/send_email.bal [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        // Body content (text) of the email is added as follows.
+        // This field is optional.
+        body: "This is a sample email.",
+        // Email author's address is added as follows. This field is mandatory.
+        'from: "author@email.com",
+        // Email sender service address is added as follows.
+        // This field is optional. `Sender` is same as the `'from` when the
+        // email author himself sends the email.
+        sender: "sender@email.com",
+        // List of recipients when replying to the email is added as follows.
```
Comment (praneesha) [inline examples/send-email/send_email.bal:29]:
```suggestion
        // List of recipients when replying to the email are added as follows.
```

File: examples/send-email/send_email.description [hunk 10]
```diff
@@ -0,0 +1,6 @@
+// The Email Connector is used to send (with SMTP) emails
+// using the SSL or STARTTLS protocols. This sample
+// includes sending emails with default configurations
+// over SSL using default ports.<br/><br/>
```
Comment (praneesha) [inline examples/send-email/send_email.description:4]:
```suggestion
// over SSL using the default ports.<br/><br/>
```

File: examples/send-email/send_email.metatags [hunk 11]
```diff
@@ -0,0 +1,2 @@
+description: This is a BBE on sending emails.
```
Comment (praneesha) [inline examples/send-email/send_email.metatags:1]:
```suggestion
description: This BBE is about sending emails.
```

**Review by praneesha — APPROVED** [review]

File: examples/receive-email-using-client/receive_email_using_client.bal (hunk 1 above)
Comment (Maninda) [inline examples/receive-email-using-client/receive_email_using_client.bal:22]:
```suggestion
```

File: examples/receive-email-using-client/receive_email_using_client.bal (hunk 2 above)
Comment (Maninda) [inline examples/receive-email-using-client/receive_email_using_client.bal:43]:
```suggestion
```

File: examples/receive-email-using-client/receive_email_using_client.description (hunk 3 above)
Comment (Maninda) [inline examples/receive-email-using-client/receive_email_using_client.description:1]:
```suggestion
// The email client is used to receive (with POP3 or IMAP4) emails
```

File: examples/receive-email-using-listener/receive_email_using_listener.description (hunk 4 above)
Comment (Maninda) [inline examples/receive-email-using-listener/receive_email_using_listener.description:1]:
```suggestion
// The email listener is used to receive (with POP3 or IMAP4) emails
```

File: examples/receive-email-using-client/receive_email_using_client.bal (hunk 1 above)
Comment (Maninda) [reply examples/receive-email-using-client/receive_email_using_client.bal]: Removed.

File: examples/receive-email-using-client/receive_email_using_client.bal (hunk 2 above)
Comment (Maninda) [reply examples/receive-email-using-client/receive_email_using_client.bal]: Removed.

File: examples/receive-email-using-client/receive_email_using_client.description (hunk 3 above)
Comment (Maninda) [reply examples/receive-email-using-client/receive_email_using_client.description]: Changed.

File: examples/receive-email-using-listener/receive_email_using_listener.description (hunk 4 above)
Comment (Maninda) [reply examples/receive-email-using-listener/receive_email_using_listener.description]: Changed.

File: examples/receive-email-using-client/receive_email_using_client.bal [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    // Read the first unseen email received by the POP3 server. `()` is
+    // returned when there are no new unseen emails. In error cases, an
+    // error is returned.
+    email:Message? emailResponse = check popClient->receiveEmailMessage();
+
+    if (emailResponse is email:Message) {
+        io:println("POP client received an email.");
+        io:println("Email Subject: ", emailResponse.subject);
+        io:println("Email Body: ", emailResponse.body);
```
Comment (Maninda) [inline examples/receive-email-using-client/receive_email_using_client.bal:19]:
```suggestion
        io:println("Email Body: ", emailResponse?.body);
```

File: examples/receive-email-using-client/receive_email_using_client.bal [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+    // Read the first unseen email received by the IMAP4 server. `()` is
+    // returned when there are no new unseen emails. In error cases, an
+    // error is returned.
+    emailResponse = check imapClient->receiveEmailMessage();
+
+    if (emailResponse is email:Message) {
+        io:println("IMAP client received an email.");
+        io:println("Email Subject: ", emailResponse.subject);
+        io:println("Email Body: ", emailResponse.body);
```
Comment (Maninda) [inline examples/receive-email-using-client/receive_email_using_client.bal:39]:
```suggestion
        io:println("Email Body: ", emailResponse?.body);
```

File: examples/receive-email-using-listener/receive_email_using_listener.bal [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+// One or many services can listen to the email listener for the periodically-polled
+// emails.
+service "emailObserver" on emailListener {
+
+    // When an email is successfully received, the `onEmailMessage` method is called.
+    remote function onEmailMessage(email:Message emailMessage) {
+        io:println("POP Listener received an email.");
+        io:println("Email Subject: ", emailMessage.subject);
+        io:println("Email Body: ", emailMessage.body);
```
Comment (Maninda) [inline examples/receive-email-using-listener/receive_email_using_listener.bal:23]:
```suggestion
        io:println("Email Body: ", emailMessage?.body);
```

**Review by lafernando — APPROVED** [review]

**Pull request merged**